### PR TITLE
Sprint 13

### PR DESCRIPTION
**Pull request by codyseibert**
### 517 - Internal user: See unread messages in work queue

 - unread messages are bolded on the work queue and considered read when the document link is clicked

Screen Shot 2019-05-01 at 11 15 05 AM

### 1472 - Mobile Docket View

- making the docket view mobile friendly

Screen Shot 2019-05-01 at 11 15 51 AM

### 531 - Practitioner: Filing an Entry of Appearance or Substitution of Counsel

- ability to associate a practitioner to a case

Screen Shot 2019-05-01 at 11 16 37 AM

### 837 - Court: Time-Out due to Inactivity

- displaying a modal when the user has been inactive
- logs the user out after 1 hour of inactivity (no mouse movement or key strokes)

Screen Shot 2019-05-01 at 11 17 31 AM

### 316 - Generation of cover sheet for all uploaded documents

- generates a cover sheet for all uploaded documents

Screen Shot 2019-05-01 at 11 18 06 AM

### CODE REVIEW

**Comment by waldoj** [comment]
Here's the code review checklist, which I will keep updated as I review this sprint:

- [ ] begin by deleting any existing copy of the Git repository and cloning a fresh copy, and running `docker system prune` to remove any cached artifacts from any prior code review
- [ ] review the pull request itself, to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] stand up the site locally
	- [ ] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [ ] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [ ] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] review static code analysis results in SonarCloud
- [ ] run a security audit of dependencies (`npm audit`) in `efcms-service`, `web-client`, and `shared`, to ensure that there are no vulnerabilities that will be deployed to production (as opposed to vulnerabilities that only have an impact on the development environment)
- [ ] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, [looking for problems](/ustaxcourt/ef-cms/blob/staging/CODE_REVIEW.md#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] look at code coverage of tests in SonarCloud
	- [ ] methodically review all new tests for correctness, quality of naming
- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

**Comment by waldoj** [comment]
@codyseibert, would y'all please include the OWASP ZAP report for this sprint?

**Comment by waldoj** [comment]
To compensate for GitHub's poor handling of tracking the status of PR-level feedback, for this PR I will make `efcms-service/serverless.yml` the hapless victim atop which I will pile feedback unrelated to the file itself.

File: efcms-service/serverless.yml [hunk 1]
```diff
@@ -223,6 +223,14 @@ functions:
           method: get
           cors: true
 
+  setMessageAsRead:
```
[the remark this answers is not in the document]
Comment (sutt0n) [reply efcms-service/serverless.yml]: A newer version of JSDoc was released to support the `<React.Fragment>` shorthand, `<>` and `</>` via https://github.com/jsdoc/jsdoc/issues/1540#issuecomment-489464282.

[the remark this answers is not in the document]
Comment (waldoj) [reply efcms-service/serverless.yml]: That fixed the problems with the JSX — the latter two-thirds of the error messages. Thanks!

File: efcms-service/serverless.yml [hunk 2]
```diff
@@ -223,6 +223,14 @@ functions:
...
+    handler: src/handlers.setMessageAsReadLambda
+    events:
+      - http:
```
[the remark this answers is not in the document]
Comment (matthopson) [reply efcms-service/serverless.yml]: Fixed bug (unrelated to timeout modal) that was causing this.

File: efcms-service/serverless.yml [hunk 3]
```diff
@@ -223,6 +223,14 @@ functions:
...
+          path: v1/messages/{messageId}/read
```
[the remark this answers is not in the document]
Comment (matthopson) [reply efcms-service/serverless.yml]: Pulled the fix for this into this branch.

File: efcms-service/serverless.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (waldoj) [reply efcms-service/serverless.yml]: It looks like the fix for the return type errors isn't working:

```
$ pwd
/Users/waldojaquith/ef-cms

$ jsdoc --version
JSDoc 3.6.1 (Sun, 05 May 2019 19:24:09 GMT)

$ jsdoc -r .
ERROR: Unable to parse a tag's type expression for source file /Users/waldojaquith/Desktop/ef-cms/shared/src/business/entities/Case.js in line 646 with tag title "returns" and text "{*|[]}": Invalid type expression "*|[]": Expected "!", "$", "'", "*", ".", "...", "0", "?", "@", "Function", "\"", "\\", "_", "break", "case", "catch", "class", "const", "continue", "debugger", "default", "delete", "do", "else", "enum", "export", "extends", "false", "finally", "for", "function", "if", "implements", "import", "in", "instanceof", "interface", "let", "new", "null", "package", "private", "protected", "public", "return", "static", "super", "switch", "this", "throw", "true", "try", "typeof", "undefined", "var", "void", "while", "with", "yield", "{", Unicode letter number, Unicode lowercase letter, Unicode modifier letter, Unicode other letter, Unicode titlecase letter, Unicode uppercase letter, or [1-9] but "[" found.
ERROR: Unable to parse a tag's type expression for source file /Users/waldojaquith/Desktop/ef-cms/shared/src/business/entities/Case.js in line 650 with tag title "returns" and text "{*|[]}": Invalid type expression "*|[]": Expected "!", "$", "'", "*", ".", "...", "0", "?", "@", "Function", "\"", "\\", "_", "break", "case", "catch", "class", "const", "continue", "debugger", "default", "delete", "do", "else", "enum", "export", "extends", "false", "finally", "for", "function", "if", "implements", "import", "in", "instanceof", "interface", "let", "new", "null", "package", "private", "protected", "public", "return", "static", "super", "switch", "this", "throw", "true", "try", "typeof", "undefined", "var", "void", "while", "with", "yield", "{", Unicode letter number, Unicode lowercase letter, Unicode modifier letter, Unicode other letter, Unicode titlecase letter, Unicode uppercase letter, or [1-9] but "[" found.
ERROR: Unable to parse a tag's type expression for source file /Users/waldojaquith/Desktop/ef-cms/shared/src/business/entities/Case.js in line 717 with tag title "returns" and text "{*|[]}": Invalid type expression "*|[]": Expected "!", "$", "'", "*", ".", "...", "0", "?", "@", "Function", "\"", "\\", "_", "break", "case", "catch", "class", "const", "continue", "debugger", "default", "delete", "do", "else", "enum", "export", "extends", "false", "finally", "for", "function", "if", "implements", "import", "in", "instanceof", "interface", "let", "new", "null", "package", "private", "protected", "public", "return", "static", "super", "switch", "this", "throw", "true", "try", "typeof", "undefined", "var", "void", "while", "with", "yield", "{", Unicode letter number, Unicode lowercase letter, Unicode modifier letter, Unicode other letter, Unicode titlecase letter, Unicode uppercase letter, or [1-9] but "[" found.
ERROR: Unable to parse a tag's type expression for source file /Users/waldojaquith/Desktop/ef-cms/shared/src/business/entities/Case.js in line 721 with tag title "returns" and text "{*|[]}": Invalid type expression "*|[]": Expected "!", "$", "'", "*", ".", "...", "0", "?", "@", "Function", "\"", "\\", "_", "break", "case", "catch", "class", "const", "continue", "debugger", "default", "delete", "do", "else", "enum", "export", "extends", "false", "finally", "for", "function", "if", "implements", "import", "in", "instanceof", "interface", "let", "new", "null", "package", "private", "protected", "public", "return", "static", "super", "switch", "this", "throw", "true", "try", "typeof", "undefined", "var", "void", "while", "with", "yield", "{", Unicode letter number, Unicode lowercase letter, Unicode modifier letter, Unicode other letter, Unicode titlecase letter, Unicode uppercase letter, or [1-9] but "[" found.
```

[the remark this answers is not in the document]
Comment (waldoj) [reply efcms-service/serverless.yml]: That did it! Thanks, @sutt0n. :)
